### PR TITLE
MAT-10065: Move validateUsingStatements outside cache to ensure model…

### PR DESCRIPTION
**Pull request by etan-sb**
…/version validation always runs regardless of cache hit or miss; bump version to 3.1.4

## CQL to ELM Translation Service PR

Jira Ticket: [MAT-10065](https://jira.cms.gov/browse/MAT-10065)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/cql_elm_translator/service/CqlLibraryService.java [hunk 1]
```diff
@@ -46,6 +44,26 @@ public void setUpLibrarySourceProvider(String cql, String accessToken) {
   }
 
   public String getLibraryCql(String name, String version, String accessToken) {
+    String cql = getRawLibraryCql(name, version, accessToken);
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/cql_elm_translator/service/CqlLibraryService.java:47]: I like this refactor. It moves the validation check out of the `fetch` method, making both clearer and more single-purpose.

File: src/main/java/gov/cms/madie/cql_elm_translator/service/CqlLibraryService.java
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/cql_elm_translator/service/CqlLibraryService.java]: What's the purpose of this return? Are we expecting to run this service without the cache and need this fallback?

**Review by jkotanchik-SB — APPROVED** [review]
